test/Microsoft.AspNet.Mvc.TagHelpers.Test/LinkTagHelperTest.cs

```diff
@@ -298,12 +287,11 @@ public static TheoryData RunsWhenRequiredAttributesArePresent_Data
             helper.Process(context, output);
 
             // Assert
-            Assert.Null(output.TagName);
```

Comment:
Than make that case a separate test. Concerned we're not consistently checking what the tag helper does and does not do in all cases.